src/libs/highlevel/README.md

```diff
@@ -0,0 +1,300 @@
+# High-Level API
+
+## Introduction
+
+The goal of the high-hevel API is to increase the usability of libelektra for developers who want to integrate Elektra into their
+applications. Applications usually do not want to use low-level APIs. `KDB` and `KeySet` are useful for plugins and to implement APIs, but
+not to be directly used in applications. The high-level API should be extremely easy to get started with and at the same time it
+should be hard to use it in a wrong way. This tutorial gives an introduction for developers who want to elektrify their application
+using the high-level API.
+
+The API supports the following types, which are taken from the CORBA specification:
+
+* **String**: a string of characters, represented by `KDB_TYPE_STRING` in metadata
+* **Boolean**: a boolean value `true` or `false`, represented by `KDB_TYPE_BOOLEAN` in metadata, in the KDB the raw value `"1"` is
+               regarded, as true, any other value is considered false
+* **Char**: a single character, represented by `KDB_TYPE_CHAR` in metadata
+* **Octet**: a single byte, represented by `KDB_TYPE_OCTET` in metadata
+* **(Unsigned) Short**: a 16-bit (unsigned) integer, represented by `KDB_TYPE_SHORT` (`KDB_TYPE_UNSIGNED_SHORT`) in metadata
+* **(Unsigned) Long**: a 32-bit (unsigned) integer, represented by `KDB_TYPE_LONG` (`KDB_TYPE_UNSIGNED_LONG`) in metadata
+* **(Unsigned) Long Long**: a 64-bit (unsigned) integer, represented by `KDB_TYPE_LONG_LONG` (`KDB_TYPE_UNSIGNED_LONG_LONG`) in metadata
+* **Float**: whatever your compiler treats as `float`, probably IEEE-754 single-precision, represented by `KDB_TYPE_FLOAT` in metadata
+* **Double**: whatever your compiler treats as `double`, probably IEEE-754 double-precision, represented by `KDB_TYPE_DOUBLE` in metadata
+* **Long Double**: whatever your compiler treats as `long double`, not always available, represented by `KDB_TYPE_LONG_DOUBLE` in metadata
+
+## Setup
+
+First you have to add `elektra-highlevel` to the linked libraries of your application. To be able to use it in your source file,
```

Comment:
Is this enough (no other lib to link against needed?)? Can you please create an example programs similar to the ones in examples/external